### ISSUE TITLE
model_factory __init__ missing 

### ISSUE THREAD

**Issue by SaqibMamoon** [bishalth01/DSAM#1]
Hi, Thankyou for sharing the code, and comparison models along with it. 

However, the model is generating error, certainly something is missing. model_factory fucntion.

`   from models import model_factory
ImportError: cannot import name 'model_factory' from 'models' (unknown location)
`

Thank you.

**Comment by SaqibMamoon** [comment]
It feels like there are multiple components of the model missing, if i'm correct.

in the DSAM-master/dataset/dataloader.py the error of `init_dataloader_with_edges` function, with no dataset.data.y available. 

In addition to the first issue reported, could you please also provide additional comments, how to successfully run the model on ABIDE data, along with the settings. It would be a great help.

Thank you very much.

**Comment by SaqibMamoon** [comment]
@bishalth01 This is a kind reminder to upload the missing files/portions. Thanks a lot.

**Comment by jessessempijja** [comment]
Hi @SaqibMamoon , were you able to get this model to run?

**Comment by SaqibMamoon** [comment]
> Hi [@SaqibMamoon](https://github.com/SaqibMamoon) , were you able to get this model to run?

No, I couldn't, the missing part if the most important one, to hand the data flow among the files. I am still waiting if the authors share the complete code.

**Comment by bishalth01** [comment]
Thanks for the comments. I have added missing __init__ function in models.

Please note that loading the data might be different (assuming we all work with different kinds of data and it cannot be publicly presented), so I cannot be sure of dataloaders and cannot help exactly as it depends upon how data is stored in the repository. 

Please load the data in any way you want (with timeseries, FC and labels), and you can refer this file: 

https://github.com/bishalth01/DSAM/blob/master/models/DSAM/dsam_interpretable.py which should contain most components of DSAM (perfectly working code). The forward pass should contain all details. 

Feel free to ask any queries related to paper and method, thanks for the engagement.

**Comment by SaqibMamoon** [comment]
> Thanks for the comments. I have added missing **init** function in models.
> 
> Please note that loading the data might be different (assuming we all work with different kinds of data and it cannot be publicly presented), so I cannot be sure of dataloaders and cannot help exactly as it depends upon how data is stored in the repository.
> 
> Please load the data in any way you want (with timeseries, FC and labels), and you can refer this file:
> 
> https://github.com/bishalth01/DSAM/blob/master/models/DSAM/dsam_interpretable.py which should contain most components of DSAM (perfectly working code). The forward pass should contain all details.
> 
> Feel free to ask any queries related to paper and method, thanks for the engagement.

Thank you for the reply and updating the repo. I have been able to input the data, but There are still some issues, such as how edge_index and edge_attr were calculated and I couldn't find a code to include these to "data", the section of code from file "braingnndsamtraining.py" is given below to illustrate.

`for data in self.train_dataloader:
            batch_size = len(data.y)
            pseudo = self.identity_matrix.unsqueeze(0).expand(batch_size, -1, -1).reshape(-1, self.identity_matrix.shape[-1])  # Get directly from GPU
            data.pos = pseudo

            self.current_step += 1
            lr_scheduler.update(optimizer=optimizer, step=self.current_step)

            time_series = data.x.cuda(non_blocking=True)
            node_feature = data.pearson_corr  # if not used on GPU, keep it on CPU
            # node_feature = torch.tensor(data.pearson_corr, dtype=torch.float16).cuda(non_blocking=True)  # if not used on GPU, keep it on CPU
            label = data.y.cuda(non_blocking=True)
            edge_index = data.edge_index.cuda(non_blocking=True)
            edge_attr = data.edge_attr.cuda(non_blocking=True)
            pseudo_torch = data.pos
            batch = data.batch.cuda(non_blocking=True)
`

How and were should I get the data ( edge_index, edge, attr, and pos, etc ). could you please further details and clarification. Thank you very much.
